app/lib/discourse_automation/script.rb

```diff
@@ -0,0 +1,88 @@
+module DiscourseAutomation
+  class Script
+    include DiscourseAutomation::ScriptDSL
+
+    attr_reader :automation
+
+    def initialize(automation)
+      @automation = automation
+    end
+
+    def self.add_script(name, &block)
+      define_method("script_#{name}", &block)
+    end
+
+    def self.all
+      DiscourseAutomation::Script
```

Comment:
Not sure how fast/slow `instance_methods.grep` is, but we might want to cache the result and only clear it when one calls `add_script`?